code/modules/power/supermatter/supermatter.dm

```diff
@@ -413,21 +420,41 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		n2ocomp += clamp(max(removed.gases[/datum/gas/nitrous_oxide][MOLES]/combined_gas, 0) - n2ocomp, -1, gas_change_rate)
 		n2comp += clamp(max(removed.gases[/datum/gas/nitrogen][MOLES]/combined_gas, 0) - n2comp, -1, gas_change_rate)
+		h2ocomp += clamp(max(removed.gases[/datum/gas/water_vapor][MOLES]/combined_gas, 0) - h2ocomp, -1, gas_change_rate)
 
 		//We're concerned about pluoxium being too easy to abuse at low percents, so we make sure there's a substantial amount.
 		if(pluoxiumcomp >= 0.15)
 			pluoxiumbonus = 1    //makes pluoxium only work at 15%+
 		else
 			pluoxiumbonus = 0
 
+		if (h2ocomp >= 0.35 && h2ocomp <= 0.45) //heat protection from h2o only works when it's between 35 and 45%
+			h2obonus = 1
+		else
+			h2obonus = 0
+
+		if (h2ocomp >= 0.15 && h2ocomp <= 0.45) //the engine will stop producing power from 15% to 45% h2o (slow down the o2 and plasma generation too)
+			h2omalus = 0.05
+		else if (h2ocomp < 0.05)
+			h2omalus = 1
+		else
+			h2omalus = 2  //when gas comp is above 0.45 or between 0.15 and 0.05, the engine will start to freak out (may need test on the amount of freak out)
+
+		if (h2ocomp < 0.30) //variability in the h2o penalty calculation depending on the %
+			h2ofixed = 1
+		else if (h2ocomp >= 0.30 && h2ocomp <= 0.50)
+			h2ofixed = 0.5
+		else
+			h2ofixed = 2
+
 		//No less then zero, and no greater then one, we use this to do explosions and heat to power transfer
-		gasmix_power_ratio = min(max(plasmacomp + o2comp + co2comp + tritiumcomp + bzcomp - pluoxiumcomp - n2comp, 0), 1)
-		//Minimum value of 1.5, maximum value of 15
-		dynamic_heat_modifier = max((plasmacomp * PLASMA_HEAT_PENALTY) + (o2comp * OXYGEN_HEAT_PENALTY) + (co2comp * CO2_HEAT_PENALTY) + (tritiumcomp * TRITIUM_HEAT_PENALTY) + ((pluoxiumcomp * PLUOXIUM_HEAT_PENALTY) * pluoxiumbonus) + (n2comp * NITROGEN_HEAT_PENALTY) + (bzcomp * BZ_HEAT_PENALTY), 0.5)
-		//Value between 6 and 1
-		dynamic_heat_resistance = max((n2ocomp * N2O_HEAT_RESISTANCE) + ((pluoxiumcomp * PLUOXIUM_HEAT_RESISTANCE) * pluoxiumbonus), 1)
-		//Value between 30 and -5, used to determine radiation output as it concerns things like collecters
-		power_transmission_bonus = (plasmacomp * gas_trans["PL"]) + (o2comp * gas_trans["O2"]) + (bzcomp * gas_trans["BZ"]) + (tritiumcomp * gas_trans["TRIT"]) + ((pluoxiumcomp * gas_trans["PLX"]) * pluoxiumbonus)
+		gasmix_power_ratio = min(max(((plasmacomp + o2comp + co2comp + h2ocomp + tritiumcomp + bzcomp - pluoxiumcomp - n2comp) * h2omalus), 0), 1)
+		//Minimum value of 1.5, maximum value of 27.8 (again changed and i don't know how to calculate it)
+		dynamic_heat_modifier = max(((plasmacomp * PLASMA_HEAT_PENALTY) + (o2comp * OXYGEN_HEAT_PENALTY)) * h2omalus + (h2ocomp * H2O_HEAT_PENALTY) * h2ofixed + (co2comp * CO2_HEAT_PENALTY) + (tritiumcomp * TRITIUM_HEAT_PENALTY) + ((pluoxiumcomp * PLUOXIUM_HEAT_PENALTY) * pluoxiumbonus) + (n2comp * NITROGEN_HEAT_PENALTY) + (bzcomp * BZ_HEAT_PENALTY), 0.5)
```

Comment:
```suggestion
		//Minimum value of 1.5, maximum value of 23
		dynamic_heat_modifier = max(((plasmacomp * PLASMA_HEAT_PENALTY) + (o2comp * OXYGEN_HEAT_PENALTY)) * h2omalus + (h2ocomp * H2O_HEAT_PENALTY) * h2ofixed + (co2comp * CO2_HEAT_PENALTY) + (tritiumcomp * TRITIUM_HEAT_PENALTY) + ((pluoxiumcomp * PLUOXIUM_HEAT_PENALTY) * pluoxiumbonus) + (n2comp * NITROGEN_HEAT_PENALTY) + (bzcomp * BZ_HEAT_PENALTY), 0.5)
```
I kinda understand what you're doing here but it seems like too much clutter, could you explain the interaction in depth?

Reply:
I changed the value and moved the h2omalus to be applied to all the comp except the h2o. What's happening here is that h2o will lower the heat production of all other gases to 5% of their original value when between 15% and 45% of the composition in the SM, but it will not lower its own heat production untill h2ofixed reaches the desired amount (between 30% and 50%), that way the SM is safely usable to produce heat instead of radiations.